### ISSUE TITLE
panic in pgdriver.(*reader).ReadN

### ISSUE THREAD

**Issue by l0nax** [uptrace/bun#73]
The following panic occures:
```
panic [/root/.gvm/gos/go1.16/src/runtime/panic.go:106]: runtime error: slice bounds out of range [:130] with capacity 64

panic(0xdd4920, 0xc000152ac8)
        /root/.gvm/gos/go1.16/src/runtime/panic.go:965 +0x1b9
github.com/uptrace/bun/driver/pgdriver.(*reader).ReadN(0xc0002e8020, 0x82, 0x0, 0x18, 0x4, 0x18, 0xc0004910d8)
        /root/.go/pkg/mod/github.com/uptrace/bun/driver/pgdriver@v0.3.0/proto.go:93 +0xf0
github.com/uptrace/bun/driver/pgdriver.readBytesCol(0xc0002e8020, 0x82, 0x93cfb8, 0xc0002e8020, 0x4, 0xc00016c100)
        /root/.go/pkg/mod/github.com/uptrace/bun/driver/pgdriver@v0.3.0/column.go:121 +0x5f
github.com/uptrace/bun/driver/pgdriver.readColumnValue(0xc0002e8020, 0xc000000011, 0x82, 0x0, 0xc000128168, 0x0, 0x0)
        /root/.go/pkg/mod/github.com/uptrace/bun/driver/pgdriver@v0.3.0/column.go:58 +0x1b4
github.com/uptrace/bun/driver/pgdriver.(*rows).readDataRow(0xc000128120, 0xc0002e8020, 0xc00023c0a0, 0x5, 0x5, 0x40f65b, 0xc00023c0a0)
        /root/.go/pkg/mod/github.com/uptrace/bun/driver/pgdriver@v0.3.0/driver.go:484 +0xff
github.com/uptrace/bun/driver/pgdriver.(*rows).next(0xc000128120, 0xc00023c0a0, 0x5, 0x5, 0x0, 0xc000491280, 0x44fa8c)
        /root/.go/pkg/mod/github.com/uptrace/bun/driver/pgdriver@v0.3.0/driver.go:437 +0x1db
github.com/uptrace/bun/driver/pgdriver.(*rows).Next(0xc000128120, 0xc00023c0a0, 0x5, 0x5, 0xc000491340, 0x720bde)
        /root/.go/pkg/mod/github.com/uptrace/bun/driver/pgdriver@v0.3.0/driver.go:414 +0x5a
database/sql.(*Rows).nextLocked(0xc000210480, 0xc0003e0000)
        /root/.gvm/gos/go1.16/src/database/sql/sql.go:2865 +0xbf
database/sql.(*Rows).Next.func1()
        /root/.gvm/gos/go1.16/src/database/sql/sql.go:2843 +0x3c
database/sql.withLock(0xfec220, 0xc0002104b0, 0xc000491380)
        /root/.gvm/gos/go1.16/src/database/sql/sql.go:3294 +0x69
database/sql.(*Rows).Next(0xc000210480, 0x0)
        /root/.gvm/gos/go1.16/src/database/sql/sql.go:2842 +0x87
github.com/uptrace/bun.(*structTableModel).ScanRows(0xc0001a0300, 0xff3290, 0xc000130038, 0xc000210480, 0x88, 0x0, 0x0)
        /root/.go/pkg/mod/github.com/uptrace/bun@v0.3.0/model_table_struct.go:233 +0x2f
github.com/uptrace/bun.(*baseQuery).scan(0xc000316d00, 0xff3290, 0xc000130038, 0xfe4600, 0xc000316d00, 0xc0003ea000, 0x88, 0xfeb7f8, 0xc0001a0300, 0x1, ...)
        /root/.go/pkg/mod/github.com/uptrace/bun@v0.3.0/query_base.go:448 +0x1b2
github.com/uptrace/bun.(*SelectQuery).Scan(0xc000316d00, 0xff3290, 0xc000130038, 0x0, 0x0, 0x0, 0x40fdb8, 0x40)
        /root/.go/pkg/mod/github.com/uptrace/bun@v0.3.0/query_select.go:679 +0x21b
gitlab.fabmation.info/deserve/server/models.GetFileByAssetID(0x700, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0003721b0, ...)
        /root/.go/src/gitlab.fabmation.info/deserve/server/models/file.go:77 +0x13d
```

with the given code snippet:
```go
package models

// File is a single raw file of an update asset.
type File struct {
	bun.BaseModel `bun:"file"`

	ID            int64         `bun:"type:integer"`
	UpdateAssetID int64         `bun:"type:integer"`
	S3Path        string        `bun:"type:text"`
	HashType      HashAlgorithm `bun:"type:hash_algorithm"`
	Hash          Hash          `bun:"type:bytea"`
}

func GetFileByAssetID(asset int64) (File, error) {
	f := File{ID: asset}

	err := db.NewSelect().
		Model(&f).
		WherePK().
		Scan(context.Background())

	return f, err
}
```

It is caused because the buffer `buf` has been allocated with `64` bytes:
https://github.com/uptrace/bun/blob/b33abf83e306ca38ed84313e8882a3e7b4159c9d/driver/pgdriver/proto.go#L85-L98

In the `Discard` method, it's handled:
https://github.com/uptrace/bun/blob/b33abf83e306ca38ed84313e8882a3e7b4159c9d/driver/pgdriver/proto.go#L100-L104

The specific row (`Hash`) stores a SHA512 hash, i.e. `d688b693004f04e20736df27dd1d3486ad62f417203414cd4c57b39d6889f8da3603d1bb224a8e8a8d55e05d17b3d1ba1fad1406de199144e68d05ba2fadff07` (len: `128`). The additional two bytes are because of DB internals of, how to store the `bytea` row.

**Comment by l0nax** [comment]
The reader was implemented 15 days ago. I have tested the given code snipped against the latest version (`v0.3.0`).

/cc @vmihailenco

**Comment by vmihailenco** [comment]
Thanks for the report. Should be fixed in v0.3.1
